packages/variant-theme/package.json

```diff
@@ -2,13 +2,17 @@
   "name": "@fluentui-contrib/variant-theme",
   "version": "0.0.1",
   "dependencies": {
-    "@swc/helpers": "~0.5.2"
+    "@swc/helpers": "~0.5.2",
+    "@fluentui/react-components": ">=9.46.3 <10.0.0",
```

Comment:
react-components should never be a direct dependency of extension, please remove this

Reply:
I have a question about the dependency rule: Which dependencies should be direct dependencies, and which should be peer dependencies? From my understanding, packages used directly should be direct dependencies. Since theme generator is using something from @fluentui/react-components, I set it as a direct dependency.

Reply:
so for our case deps should look like this:

```json
{
"dependencies": {
    "@swc/helpers": "~0.5.2",
    "@fluentui/scheme-utilities": "^8.3.57",
    "@fluentui/react-migration-v8-v9": "^9.6.22",
    "@fluentui/react": "^8.119.3"
  },
  "peerDependencies": {
    "@fluentui/react-components": ">=9.46.3 <10.0.0",
    "@types/react": ">=16.8.0 <19.0.0",
    "@types/react-dom": ">=16.8.0 <19.0.0",
    "react": ">=16.8.0 <19.0.0",
    "react-dom": ">=16.8.0 <19.0.0"
  }
}
```

- @fluentui/react-migration-v8-v9 and @fluentui/react version range will always need to be aligned in this pacakge in order to not create dupes
- whilst @fluentui/react-components ships as dependency of react-migration-v8-v9 ( which is a bug ), lets follow our patterns for v9 extensions and keep it as peerDependency

Reply:
have changed package.json as your comment